airbyte-integrations/connectors/destination-s3/src/main/java/io/airbyte/integrations/destination/s3/util/AvroRecordHelper.java

```diff
@@ -52,32 +47,4 @@ public static JsonNode pruneAirbyteJson(final JsonNode input) {
     return output;
   }
 
-  public static void obtainPaths(String currentPath, JsonNode jsonNode, Map<JsonNode, String> jsonNodePathMap) {
-    if (jsonNode.isObject()) {
-      ObjectNode objectNode = (ObjectNode) jsonNode;
-      Iterator<Map.Entry<String, JsonNode>> iter = objectNode.fields();
-      String pathPrefix = currentPath.isEmpty() ? "" : currentPath + "/";
-      String[] pathFieldsArray = currentPath.split("/");
-      String parent = Arrays.stream(pathFieldsArray)
-          .filter(x -> !x.equals("items"))
-          .filter(x -> !x.equals("properties"))
-          .filter(x -> !x.equals(pathFieldsArray[pathFieldsArray.length - 1]))
-          .collect(Collectors.joining("."));
-      if (!parent.isEmpty()) {
-        jsonNodePathMap.put(jsonNode, parent);
-      }
-      while (iter.hasNext()) {
-        Map.Entry<String, JsonNode> entry = iter.next();
-        obtainPaths(pathPrefix + entry.getKey(), entry.getValue(), jsonNodePathMap);
-      }
-    } else if (jsonNode.isArray()) {
-      ArrayNode arrayNode = (ArrayNode) jsonNode;
-
-      for (int i = 0; i < arrayNode.size(); i++) {
-        String arrayPath = currentPath + "/" + i;
-        obtainPaths(arrayPath, arrayNode.get(i), jsonNodePathMap);
-      }
-    }
-  }
```

Comment:
This method proactively calculate all paths to generate the namespaces. However, this is not necessary. The namespaces can be computed while travesing the Json schema.